core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3231,4 +3231,42 @@ private static int unsetLocalDictForComplexColumns(List<ColumnSchema> allColumns
     }
     return columnLocalDictGenMap;
   }
+
+  /**
+   * This method get the carbon file format version
+   *
+   * @param carbonTable
+   * carbon Table
+   */
+  public static ColumnarFormatVersion getFormatVersion(CarbonTable carbonTable) throws IOException
+  {
+    String tablePath = carbonTable.getTablePath();
```

Comment:
This check would be wrong, carbon files not always be under table path directly.